newIDE/app/src/VariablesList/index.js

```diff
@@ -237,27 +280,49 @@ export default class VariablesList extends React.Component<Props, State> {
 
   _renderEmpty() {
     return (
-      <div>
-        <EmptyMessage
-          style={styles.emptyExplanation}
-          messageStyle={styles.emptyExplanationMessage}
-        >
-          {this.props.emptyExplanationMessage || ''}
-        </EmptyMessage>
-        <EmptyMessage
-          style={styles.emptyExplanation}
-          messageStyle={styles.emptyExplanationMessage}
-        >
-          {this.props.emptyExplanationSecondMessage}
-        </EmptyMessage>
-      </div>
+      !!this.props.emptyExplanationMessage && (
+        <div>
+          <EmptyMessage
+            style={styles.emptyExplanation}
+            messageStyle={styles.emptyExplanationMessage}
+          >
+            {this.props.emptyExplanationMessage}
+          </EmptyMessage>
+          <EmptyMessage
+            style={styles.emptyExplanation}
+            messageStyle={styles.emptyExplanationMessage}
+          >
+            {this.props.emptyExplanationSecondMessage}
+          </EmptyMessage>
+        </div>
+      )
     );
   }
 
   render() {
-    const { variablesContainer } = this.props;
+    const { variablesContainer, inheritedVariablesContainer } = this.props;
     if (!variablesContainer) return null;
 
+    // map all object variables, if they have been passed
+    const containerObjectVariablesTree = inheritedVariablesContainer
+      ? mapFor(0, inheritedVariablesContainer.count(), index => {
+          const name = inheritedVariablesContainer.getNameAt(index);
+          if (this._getInstanceVariableType(name) === 'object') {
```

Comment:
Not sure if we need to call `_getInstanceVariableType ` here.
We're iterating on the parent (i.e: object) variables. So we only need to do a `if (variablesContainer.has(name)) { ... }` to know if the variable is being replaced in the variables container.

Reply:
should be` if(!variablesContainer.has(name)){...} ` but ok :)

Reply:
Yep right!